packages/core/src/external/aws/lambda-logic/bundle-to-brief.ts

```diff
@@ -1,128 +1,746 @@
+    prompt: SUMMARY_PROMPT as any, // eslint-disable-line @typescript-eslint/no-explicit-any
+  });
+
+  // this is the prompt for combining the summaries into a single paragraph
+  const summaryTemplateRefined = `
+${systemPrompt}
+
+Today's date is ${todaysDate}.
+Your goal is to write a summary of the patient's most recent medical history, so that another doctor can understand the patient's medical history to be able to treat them effectively.
+Here are the previous summaries written by you of sections of the patient's medical history:
+--------
+{text}
+--------
+
+Combine these summaries into a single, comprehensive summary of the patient's most recent medical history in a single paragraph.
+
+Don't tell me that you are writing a summary, just write the summary. Also, don't tell me about any limitations of the information provided.
+
+SUMMARY:
+`;
+  const SUMMARY_PROMPT_REFINED = PromptTemplate.fromTemplate(summaryTemplateRefined);
+  const summaryChainRefined = new StuffDocumentsChain({
+    llmChain: new LLMChain({
+      llm: llmSummary as any, // eslint-disable-line @typescript-eslint/no-explicit-any
+      prompt: SUMMARY_PROMPT_REFINED as any, // eslint-disable-line @typescript-eslint/no-explicit-any
+    }),
+    documentVariableName: "text",
+  });
+
+  const mapReduce = new MapReduceDocumentsChain({
+    llmChain: summaryChain,
+    combineDocumentChain: summaryChainRefined,
+    documentVariableName: "text",
+    verbose: true,
+  });
+
+  const summary = (await mapReduce.invoke({
+    input_documents: docs,
+  })) as { text: string };
+
+  if (!summary.text) return undefined;
+  return summary.text;
 }
 
-export async function bundleToBrief(
-  fhirBundle: Bundle,
-  cxId: string,
-  patientId: string
-): Promise<string | undefined> {
-  const { log } = out(`AI Brief for cx ${cxId}, patient ${patientId}`);
-  let processedBundleInput = await prepareBundleForBrief(fhirBundle);
-  if (!processedBundleInput) return undefined;
-
-  if (processedBundleInput.length > MAXIMUM_BRIEF_STRING_LENGTH) {
-    const initialLength = processedBundleInput.length;
-    processedBundleInput = processedBundleInput.slice(0, MAXIMUM_BRIEF_STRING_LENGTH);
-    const msg = `Brief string input was truncated`;
-    log(`${msg}. Initial length was: ${initialLength}`);
-    capture.message(msg, {
-      extra: {
-        patientId,
-        cxId,
-        stringLength: processedBundleInput.length,
-        maximumLength: MAXIMUM_BRIEF_STRING_LENGTH,
-      },
-      level: "info",
+export function prepareBundleForBrief(bundle: Bundle): string | undefined {
+  if (bundle.entry?.length === 0) return undefined;
+  const filteredString = filterBundleResources(bundle);
+  return filteredString;
+}
+
+function filterBundleResources(bundle: Bundle) {
+  const patient = findPatientResource(bundle);
+  if (!patient) return undefined;
+
+  // First pass to remove a ton of useless stuff and apply resource-specific modifications
```

Comment:
Most of the code on this PR is just reverting a revert, these included. I see your point, but I'd like to avoid changing things that we're bringing from a `git revert`

Reply:
not applicable any longer